src/app/cli/src/commands/compact_command.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use kamu::domain::compact_service::CompactService;
+use kamu::domain::{
+    DatasetRepository,
+    VerificationMultiListener,
+    VerificationOptions,
+    VerificationService,
+};
+use opendatafabric::{DatasetHandle, DatasetRef};
+
+use crate::{CLIError, Command, CompactionMultiProgress, VerificationMultiProgress};
+
+pub struct CompactCommand {
+    dataset_repo: Arc<dyn DatasetRepository>,
+    verification_svc: Arc<dyn VerificationService>,
+    compact_svc: Arc<dyn CompactService>,
+    dataset_ref: DatasetRef,
+    dataset_dir_path: PathBuf,
+}
```

Comment:
We discussed the importance of limiting maximum data chunks. 
Data could be bigger than the entire memory of the computer running it.
Note that deployment API services have quite limited memory, usually not more than 1 GB.

Reply:
One could say that our querying is also not prepared for such large files. I'll give another point - without chunks we are limiting transfer parallelism. I guess it would be much faster to have N parallel downloads of a few-megabyte files than 1 huge file.

Reply:
Added